module1/homework/grayscaleImages/compression.cpp

```diff
@@ -0,0 +1,60 @@
+#include "compression.hpp"
+
+#include <algorithm>
+#include <iostream>
+#include <utility>
+
+std::vector<std::pair<uint8_t, uint8_t>> compressGrayscale(
+    const std::array<std::array<uint8_t, width>, height>& bitmap) {
+    std::vector<std::pair<uint8_t, uint8_t>> compressed{};
+    compressed.reserve(height * width);
+    int occurrences = 0;
+    uint8_t color_code = 0;
+    for (const auto& row : bitmap) {
+        color_code = *row.begin();
+        occurrences = 0;
+        for (auto element : row) {
+            if (element == color_code) {
+                ++occurrences;
+            } else {
+                compressed.emplace_back(std::make_pair(color_code, occurrences));
+                color_code = element;
+                occurrences = 1;
+            }
+        }
+        compressed.emplace_back(std::make_pair(color_code, occurrences));
```

Comment:
if you rearrange loop, you will be albe to use emplace_back just once, also since new standards in c++ you can use {} to make pair instead of std::make_pair

Reply:
We managed to change that - we use only one emplace_back. I forgot that I can use curly braces instead of std::make_pair, thanks :)